enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,253 @@
+which is defaulted to nil if absent. A nil executor subject indicates that the 
+work agent doesn't perform any impersonation before requesting the hosting 
+managed cluster.
+
+
+A sample of the new `executorSubject` field in Yaml is shown below:
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata: ...
+spec:
+  executorSubject:
+    type: [ServiceAccount] # union discriminator for extensibility
+    serviceAccount:
+      name: ...
+      namespace: ...
+  workload:
+    manifests: [...]
+```
+
+### Work-agent client permission muzzling
+
+The supported executor subject types will be merely "ServiceAccount".
+For "ServiceAccount" type, intuitively the referenced service account is 
+working like an "ambassador" identity, and the work agent will need to 
+explicitly check whether that "ambassador" has sufficient RBAC permission 
+in the managed cluster via SubjectAccessReview api. Due to the limit of 
+the SubjectAccessReview api, we need to send (# of contained objects) times 
+SAR requests to the spoke cluster, but this should be fine b/c the work-agent 
+is working asynchronously. 
+
+Note that the authorization via subject access review api doesn't require the
+service account to be actually present in the managed cluster, so in the cases
+where you have too many executor subject to manage, the number of service 
+accounts resources in the managed cluster is bounded.
+
+A sample of the requesting `SubjectAccessReview` payload is:
+
+```yaml
+apiVersion: authentication.open-cluster-management.io/v1alpha1
+kind: SubjectAccessReview
+spec:
+  # requester is the referenced service-account in executor subject
+  user: "system:serviceaccount:<sa namespace>:<sa name>"
+  groups: ["system:serviceaccounts", "system:authenticated"]
+  # requesting resource attributes
+  resourceAttributes:
+    # verb
+    verb: <the intended verb, can be create/update/patch/delete>
+    # the GVR tuple
+    group: <api group>
+    version: <api version>
+    resource: <api resource name>
+    # the object identifier
+    name: <resource name>
+    namespace: <resource namespace if applicable>
+```
+
+### Architecture
+
+The overall architecture is shown in the picture below:
+
+![Architecture](./arch.png "Architecture")
+
+#### Workflow
+
+The following is what will happen when a `ManifestWork` with non-nil executor
+subject specified is applied to the hub cluster:
+
+0. Create a "ManifestWork" with non-nil executor subject
+1. The work-agent observes the new resource via watch event
+2. The work-agent builds a SubjectAccessReview request which:
+   - The target subject is the executor subject from `ManifestWork`
+   - The target attribute should contain the "*" verbs for all the inlined
+     objects in the spec of `ManifestWork`. Note that the "resourceNames"
+     should also be specified.
+3. The work-agent writes the objects to the managed cluster if SAR passes.
```

Comment:
Followup, will deletion be prevented if identity cannot delete or is cleanup immune?  Thinking about the options, I think you must honor the identity restrictions on deletion to avoid someone placing an item and then removing it to trigger a deletion they cannot manage.  You should specifically enumerate this.

In addition, perhaps you should avoid creating a thing if the identity cannot delete it on cleanup.

Reply:
@deads2k am leaning on option 1. imagine if we're deploying a combo of deployment+service in a work object, and in an update i updated the mounting service-account for that deployment from "default" to a "foo", then we will run into failure due to insufficient permission for writing service-account. in this case if we allow partial updates, the updated deployment will be constantly failing b/c that "foo" SA doesnt exist. generally i think even tho the atomic strategy is a bit clumzy but it will save some problems from happening. i clarified that in the new commit

Reply:
> In addition, perhaps you should avoid creating a thing if the identity cannot delete it on cleanup.

@deads2k i also clarified the deletion sequence in the appended commit. short version: the deletion flow will also work with an atomic strategy. the work-agent wont delete any objects until the executor subject matches all the pre-conditions.